### PR TITLE
Feature/create service

### PR DESCRIPTION
**Pull request by caiocacastro**
Give your opinion on the code, please, mister master frienderino.

### CODE REVIEW

File: Dockerfile [hunk 1]
```diff
@@ -0,0 +1,68 @@
...
+RUN pip install gdal==$(gdal-config --version) --global-option=build_ext --global-option="-I/usr/include/gdal"
+
+# Install dependencies
+COPY requirements.txt .
+RUN pip install --no-cache-dir -r requirements.txt
+
+# Copy local code to the container image.
+COPY . .
+
+# Entrypoint configuration
```
Comment (dagnaldo) [inline Dockerfile]: Remove files and comments

File: core/serializers.py [hunk 2]
```diff
@@ -0,0 +1,55 @@
+from django.contrib.gis import geos
+from rest_framework_gis import serializers as gis_serializers
+
+from .models import Detection
+import json
```
Comment (dagnaldo) [inline core/serializers.py]: Following style [guide](https://www.python.org/dev/peps/pep-0008/#imports), imports should be grouped in the following order:
- Standard library imports.
- Related third party imports.
- Local application/library specific imports.

File: core/models.py [hunk 3]
```diff
@@ -0,0 +1,94 @@
...
+
+    dt_cadastro = models.DateTimeField()
+
+    geometry = models.GeometryField()
+
+    def __str__(self):
+        return self.no_estagio
+
+    def get_es_insertion_line(self):
+        """Method for generating create string line for bulk
```
Comment (dagnaldo) [inline core/models.py]: missing period.

File: core/models.py [hunk 4]
```diff
@@ -0,0 +1,94 @@
...
+                getattr(self, f.get_attname())
+            ) for f in self._meta.get_fields() if f.get_attname() != '_id']
+
+        es_data_header = f'{{ "create": {{ "_id": "{self._id}"}}}}'
+        es_data_line = ", ".join(fields)
+
+        return f'{es_data_header}\n{{{es_data_line}}}\n'
+
+    def _format_data(self, field: str, value: object):
+        """Method for validating each specific field with each specific type
```
Comment (dagnaldo) [inline core/models.py]: Docstrings first line must be a small method defs and a second line is used to explain it.

File: core/serializers.py [hunk 5]
```diff
@@ -0,0 +1,55 @@
...
+
+
+class DetectionSerializer(gis_serializers.GeoFeatureModelSerializer):
+    """Serializer for Detection Model"""
```
Comment (dagnaldo) [inline core/serializers.py]: Mssing period.

File: core/serializers.py [hunk 6]
```diff
@@ -0,0 +1,55 @@
...
+            'nu_ponto',
+            'dt_t_zero',
+            'dt_t_um',
+            'nu_area_km2',
+            'nu_area_ha',
+            'dt_cadastro',
+        )
+
+    def unformat_geojson(self, feature: object) -> object:
+        """Method that formats features of uploaded geojson
```
Comment (dagnaldo) [inline core/serializers.py]: missing period

File: core/tests/recipes.py [hunk 7]
```diff
@@ -0,0 +1,16 @@
+from core.models import BasicElasticStructure, Detection
+from model_mommy.recipe import Recipe
+
+
+class Recipes:
+    """Mommy recipes used on test cases"""
```
Comment (dagnaldo) [inline core/tests/recipes.py]: Missing period.

File: core/tests/test_views.py [hunk 8]
```diff
@@ -0,0 +1,103 @@
...
+        """Setp up test function."""
+        self.client = APIClient()
+        self.upload_url = reverse('upload-detection')
+        self.create_url = reverse('create-detection')
+        self.delete_url = reverse('delete-detection')
+        self.recipes = Recipes()
+
+    @classmethod
+    def tearDownClass(cls):
+        """Clear index test inside ElasticSearch server"""
```
Comment (dagnaldo) [inline core/tests/test_views.py]: MIssing period.

File: core/tests/test_views.py [hunk 9]
```diff
@@ -0,0 +1,103 @@
...
+
+    def test_upload_no_file(self):
+        """Tests if no file is sent."""
+        response = self.client.post(self.upload_url)
+        self.assertTrue(status.is_client_error(response.status_code))
+
+    def test_upload_error_file(self):
+        """Tests if an json file with error is sent."""
+        self.recipes.es_object.make()
+        response = {}
```
Comment (dagnaldo) [inline core/tests/test_views.py]: Is it used for?

File: core/tests/recipes.py [hunk 10]
```diff
@@ -0,0 +1,16 @@
...
+
+    def __init__(self):
+        self.es_object = Recipe(
+            BasicElasticStructure,
+            index="detection-test",
+            identifier='Detection',
+            url='http://localhost:9200',
+            bulk_size_request=1000,
+            structure='{"aliases":{},"mappings":{"properties":{"id":{"type":"keyword"},"tb_ciclo_monitoramento_id":{"type":"keyword"},"sg_uf":{"type":"keyword"},"no_estagio":{"type":"keyword"},"no_imagem":{"type":"text","fields":{"exact":{"type":"keyword"}}},"dt_imagem":{"type":"date"},"nu_orbita":{"type":"text"},"nu_ponto":{"type":"text"},"dt_t_zero":{"type":"date"},"dt_t_um":{"type":"date"},"nu_area_km2":{"type":"double"},"nu_area_ha":{"type":"double"},"dt_cadastro":{"type":"date","format":"yyyy-MM-dd HH:mm:ss||yyyy-MM-dd||epoch_millis"},"geometry":{"type":"geo_shape"}}},"settings":{"analysis":{"filter":{"brazilian_stemmer":{"type":"stemmer","language":"brazilian"},"brazilian_stop":{"type":"stop","stopwords":"_brazilian_"}},"analyzer":{"image_analyzer":{"tokenizer":"undescore_tokenizer"}},"tokenizer":{"undescore_tokenizer":{"type":"simple_pattern_split","pattern":"_"}}}}}'
```
Comment (dagnaldo) [inline core/tests/recipes.py]: PEP8 style guide error, maximum of 79 characters.

File: core/tests/test_views.py [hunk 11]
```diff
@@ -0,0 +1,103 @@
...
+import os
+import pandas as pd
+import yaml
+
+from django.core.files.uploadedfile import SimpleUploadedFile
+from django.test import TestCase
+from django.urls import reverse
+from rest_framework.test import APIClient
+from rest_framework import status
+from .recipes import Recipes
```
Comment (dagnaldo) [inline core/tests/test_views.py]: PEP8 style guide says that local imports must be the last imports.

File: core/tests/test_views.py [hunk 12]
```diff
@@ -0,0 +1,103 @@
...
+                           for f in js['features']]
+
+        for element in series:
+            """Testing if all ids were sent to update."""
+            ids_inside_bulk_element = list(
+                filter(lambda id: id == element._id, id_feature_list))
+
+            self.assertTrue(ids_inside_bulk_element)
+
+        """Verifies if all fields inside ES Structure were serialized on the object"""
```
Comment (dagnaldo) [inline core/tests/test_views.py]: maximum of 80 characters.

File: core/tests/test_views.py [hunk 13]
```diff
@@ -0,0 +1,103 @@
...
+    def test_verify_if_file_is_serialized(self):
+        """Tests equality between serialized data and sent json file."""
+        with open('core/tests/mockData.json') as json_file:
+            js = json.loads(json_file.read())
+            series = UtilFunctions.serialize_detection_file(js)
+
+        """Verifies if returned data is a Series type"""
+        self.assertIsInstance(series, pd.Series)
+
+        """Verifies if returned data has same lenght as data sent"""
```
Comment (dagnaldo) [inline core/tests/test_views.py]: Missing period

File: core/tests/test_views.py [hunk 14]
```diff
@@ -0,0 +1,103 @@
...
+        self.assertEqual(series.size, len(js['features']))
+
+        """Verifies if returned data has same ids sent for serializing"""
```
Comment (dagnaldo) [inline core/tests/test_views.py]: Missing period

File: core/tests/test_views.py [hunk 15]
```diff
@@ -0,0 +1,103 @@
...
+        self.assertTrue(
+            'Unexpected sent json data', response.json()['msg'])
+
+    def test_verify_if_file_is_serialized(self):
+        """Tests equality between serialized data and sent json file."""
+        with open('core/tests/mockData.json') as json_file:
+            js = json.loads(json_file.read())
+            series = UtilFunctions.serialize_detection_file(js)
+
+        """Verifies if returned data is a Series type"""
```
Comment (dagnaldo) [inline core/tests/test_views.py]: Missing period

File: core/urls.py [hunk 16]
```diff
@@ -0,0 +1,27 @@
...
+Class-based views
+    1. Add an import:  from other_app.views import Home
+    2. Add a URL to urlpatterns:  path('', Home.as_view(), name='home')
+Including another URLconf
+    1. Import the include() function: from django.urls import include, path
+    2. Add a URL to urlpatterns:  path('blog/', include('blog.urls'))
+"""
+from django.urls import path
+
+from .views import (UpdateDetectionView,
```
Comment (dagnaldo) [inline core/urls.py]:
```from .views import (
    UpdateDetectionView, ClearDetectionStructure, CreateDetectionStructure
)```

File: core/urls.py [hunk 17]
```diff
@@ -0,0 +1,27 @@
...
+                    ClearDetectionStructure, CreateDetectionStructure)
+
+urlpatterns = [
+    path('detection/delete', ClearDetectionStructure.as_view(),
+         name='delete-detection'),
+    path('detection/upload', UpdateDetectionView.as_view(), name='upload-detection'),
```
Comment (dagnaldo) [inline core/urls.py]: maximum 8 characters

File: core/views.py [hunk 18]
```diff
@@ -0,0 +1,262 @@
...
+from rest_framework import generics, response, status
+
+from .serializers import DetectionSerializer
+from django.core.serializers import serialize, json
+from .models import Detection, BasicElasticStructure
+
+import pandas as pd
+import json
+import requests
+import math
```
Comment (dagnaldo) [inline core/views.py]: Please, check import guide.

File: core/views.py [hunk 19]
```diff
@@ -0,0 +1,262 @@
...
+
+from datetime import datetime
+
+now = datetime.now()
```
Comment (dagnaldo) [inline core/views.py]: Consider creating it inside of class definition.

File: core/views.py [hunk 20]
```diff
@@ -0,0 +1,262 @@
...
+
+            ClearDetectionStructure.delete(request)
+
+            CreateDetectionStructure.put(request)
+
+            insert_errors = UtilFunctions.send_bulk_list(
+                detection_series, es_structure)
+
+            if (insert_errors):
+                return response.Response({'msg': 'Some data were not inserted', 'errors': insert_errors}, status=status.HTTP_400_BAD_REQUEST)
```
Comment (dagnaldo) [inline core/views.py]: maximum 79 characters.

File: core/views.py [hunk 21]
```diff
@@ -0,0 +1,262 @@
...
+
+            return response.Response({'msg': "Created"}, status=status.HTTP_201_CREATED)
```
Comment (dagnaldo) [inline core/views.py]: maximum 79 characters.

File: core/views.py [hunk 22]
```diff
@@ -0,0 +1,262 @@
...
+        except Exception as exc:
+            return response.Response({"msg": str(exc)}, status=404)
+
+
+class ClearDetectionStructure(generics.DestroyAPIView):
+    """Clear detection view"""
+
+    def delete(request: object) -> Response:
+        """Delete method for dealing with clear ES Structure
```
Comment (dagnaldo) [inline core/views.py]: missing period

File: core/views.py [hunk 23]
```diff
@@ -0,0 +1,262 @@
...
+            if (insert_errors):
+                return response.Response({'msg': 'Some data were not inserted', 'errors': insert_errors}, status=status.HTTP_400_BAD_REQUEST)
+
+            return response.Response({'msg': "Created"}, status=status.HTTP_201_CREATED)
+        except Exception as exc:
+            return response.Response({"msg": str(exc)}, status=404)
+
+
+class ClearDetectionStructure(generics.DestroyAPIView):
+    """Clear detection view"""
```
Comment (dagnaldo) [inline core/views.py]: Missing period.

File: core/views.py [hunk 24]
```diff
@@ -0,0 +1,262 @@
...
+    def delete(request: object) -> Response:
+        """Delete method for dealing with clear ES Structure
+
+        Args:
+            request (object): Request sent for delete url
+
+        Returns:
+            Response: Response object
+        """
+        print(f'[{datetime.now() - now}] Clearing structure...')
```
Comment (dagnaldo) [inline core/views.py]: Check `now` creation.

File: core/views.py [hunk 25]
```diff
@@ -0,0 +1,262 @@
...
+        es_structure = BasicElasticStructure.objects.get(
+            identifier='Detection')
+
+        UtilFunctions.delete_es_structure(es_structure)
+
+        print(f'[{datetime.now() - now}] Clear!')
```
Comment (dagnaldo) [inline core/views.py]: Check `now` creation.

File: core/views.py [hunk 26]
```diff
@@ -0,0 +1,262 @@
...
+
+        return response.Response("Index removed", status=status.HTTP_200_OK)
+
+
+class CreateDetectionStructure(generics.UpdateAPIView):
+    """Clear detection Structure from ES Server view"""
```
Comment (dagnaldo) [inline core/views.py]: missing period.

File: core/views.py [hunk 27]
```diff
@@ -0,0 +1,262 @@
...
+    def put(request: object) -> Response:
+        """Put method for dealing with inserting a new ES Structure
+
+        Args:
+            request (object): Request sent for put url
+
+        Returns:
```
Comment (dagnaldo) [inline core/views.py]: missing period.

File: core/views.py [hunk 28]
```diff
@@ -0,0 +1,262 @@
...
+            Response: Response Object
+        """
+        print(f'[{datetime.now() - now}] Creating structure...')
```
Comment (dagnaldo) [inline core/views.py]: check `now` definition

File: core/views.py [hunk 29]
```diff
@@ -0,0 +1,262 @@
...
+        es_structure = BasicElasticStructure.objects.get(
+            identifier='Detection')
+
+        UtilFunctions.create_es_structure(es_structure)
+
+        print(f'[{datetime.now() - now}] Created!')
```
Comment (dagnaldo) [inline core/views.py]: check `now` definition

File: core/views.py [hunk 30]
```diff
@@ -0,0 +1,262 @@
...
+        Args:
+            es_structure (BasicElasticStructure): A BasicElasticStructure
+            containing all needed ES Data.
+
+        Raises:
+            ValueError: Raises error if cant insert the new structure
+                into ES Server.
+        """
+        req = requests.put(
+            f'{es_structure.url}/{es_structure.index}',
```
Comment (dagnaldo) [inline core/views.py]: Check `urllib.parse.urljoin` method. Could be useful.

File: core/views.py [hunk 31]
```diff
@@ -0,0 +1,262 @@
...
+            headers={"content-type": "application/json"},
+            json=json.loads(es_structure.structure)
+        )
+
+        if req.status_code != status.HTTP_200_OK and req.status_code != status.HTTP_404_NOT_FOUND:
```
Comment (dagnaldo) [inline core/views.py]: maximum 79 characters

File: core/views.py [hunk 32]
```diff
@@ -0,0 +1,262 @@
...
+
+        Raises:
+            ValueError: Raises error if cant delete previous structure loaded
+            into ES Server
+        """
+        req = requests.delete(
+            f'{es_structure.url}/{es_structure.index}',
+        )
+
+        if req.status_code != status.HTTP_200_OK and req.status_code != status.HTTP_404_NOT_FOUND:
```
Comment (dagnaldo) [inline core/views.py]: maximum 79 characters

File: core/views.py [hunk 33]
```diff
@@ -0,0 +1,262 @@
...
+            json=json.loads(es_structure.structure)
+        )
+
+        if req.status_code != status.HTTP_200_OK and req.status_code != status.HTTP_404_NOT_FOUND:
+            raise ValueError(
+                f'Elastic Search clearing procedure returned error. \
+                    Status code: {req.status_code} ${req.text}')
+
+    def delete_es_structure(es_structure: BasicElasticStructure):
+        """Method for deleting any existing structure for an specific
```
Comment (dagnaldo) [inline core/views.py]: Check pep8 styleguide

File: core/views.py [hunk 34]
```diff
@@ -0,0 +1,262 @@
...
+        Args:
+            es_structure (BasicElasticStructure): A BasicElasticStructure
+                containing all needed ES Data.
+
+        Raises:
+            ValueError: Raises error if cant delete previous structure loaded
+            into ES Server
+        """
+        req = requests.delete(
+            f'{es_structure.url}/{es_structure.index}',
```
Comment (dagnaldo) [inline core/views.py]: `urljoin`?

File: core/views.py [hunk 35]
```diff
@@ -0,0 +1,262 @@
...
+        )
+
+        if req.status_code != status.HTTP_200_OK and req.status_code != status.HTTP_404_NOT_FOUND:
+            raise ValueError(f'Elastic Search clearing procedure \
+                    returned error. Status code: {req.status_code} ${req.text}')
+
+    def serialize_detection_file(json_file: object) -> pd.Series:
+        """Method for serializing the uploaded json object into a Panda Series
```
Comment (dagnaldo) [inline core/views.py]: missing period.

File: core/views.py [hunk 36]
```diff
@@ -0,0 +1,262 @@
...
+        Args:
+            json_file (object): The json loaded from the json file uploaded
+
+        Raises:
+            ValueError: Raises error if there is any problem serializing the data
+
+        Returns:
+           pd.Series : Returns a new Pandas Series to be dealt further
+        """
+        print(f'[{datetime.now() - now}] serializing....')
```
Comment (dagnaldo) [inline core/views.py]: check `now`

File: core/views.py [hunk 37]
```diff
@@ -0,0 +1,262 @@
...
+        try:
+            serializer = DetectionSerializer(
+                data=json_file['features'], many=True)
+            serializer.is_valid(raise_exception=True)
+
+            print(f'[{datetime.now() - now}] Serialized!')
```
Comment (dagnaldo) [inline core/views.py]: check `now`

File: core/views.py [hunk 38]
```diff
@@ -0,0 +1,262 @@
...
+    def _create_detection_series(data: object) -> pd.Series:
+        """Creates a Panda Series with a serialized data
+
+        Args:
+            data (object): Serialized and validated Json data
+
+        Returns:
+            pd.Series: returns a Panda Series with all Detection Models
+        """
+        print(f'[{datetime.now() - now}] creating series....')
```
Comment (dagnaldo) [inline core/views.py]: check `now`

File: core/views.py [hunk 39]
```diff
@@ -0,0 +1,262 @@
...
+            object: Json Object parsed
+        """
+        try:
+            file_content = file.read()
+            return json.loads(file_content)
+        except JSONDecodeError as json_error:
+            raise ValueError(f'Unexpected sent json data')
+
+    def _create_detection_series(data: object) -> pd.Series:
+        """Creates a Panda Series with a serialized data
```
Comment (dagnaldo) [inline core/views.py]: Missing period.

File: core/views.py [hunk 40]
```diff
@@ -0,0 +1,262 @@
...
+        """
+        print(f'[{datetime.now() - now}] creating series....')
+
+        pd_detections = pd.Series(
+            [Detection(**value) for value in data])
+
+        print(f'[{datetime.now() - now}] series created!')
+        return pd_detections
+
+    def send_bulk_list(bulk_list: pd.Series, es_structure: BasicElasticStructure) -> list:
```
Comment (dagnaldo) [inline core/views.py]: Maximum 79 characters

File: core/views.py [hunk 41]
```diff
@@ -0,0 +1,262 @@
...
+
+        for chunk_id in list(range(chunks)):
+            lower_limiter = (chunk_id) * bulk_size
+            higher_limiter = (chunk_id + 1) * bulk_size
+
+            if higher_limiter >= len(bulk_list):
+                higher_limiter = len(bulk_list)
+
+            print(
+                f'[{datetime.now() - now}] sending chunk {chunk_id + 1} ({lower_limiter + 1} to {higher_limiter} elements)....')
```
Comment (dagnaldo) [inline core/views.py]: Maximum 79 characters

File: core/views.py [hunk 42]
```diff
@@ -0,0 +1,262 @@
...
+                structure with all needed data.
+
+        Raises:
+            ValueError: Raises error if there is any inserting error.
+
+        Returns:
+            list : A list containing all insertion errors.
+        """
+        print(
+            f'[{datetime.now() - now}] preparing bulk list of size {bulk_list.size}....')
```
Comment (dagnaldo) [inline core/views.py]: Maximum 79 characters

File: core/views.py [hunk 43]
```diff
@@ -0,0 +1,262 @@
...
+                higher_limiter = len(bulk_list)
+
+            print(
+                f'[{datetime.now() - now}] sending chunk {chunk_id + 1} ({lower_limiter + 1} to {higher_limiter} elements)....')
+
+            body = [t.get_es_insertion_line()
+                    for t in bulk_list[lower_limiter:higher_limiter]]
+
+            req = requests.post(
+                f'{es_structure.url}/{es_structure.index}/_bulk',
```
Comment (dagnaldo) [inline core/views.py]: `urljoin`

File: core/views.py [hunk 44]
```diff
@@ -0,0 +1,262 @@
...
+                headers={"content-type": "application/json"},
+                data="".join(body)
+            )
+
+            if req.status_code != 200:
+                raise ValueError(
+                    f'Elastic Search returned error inserting data. Status code: {req.status_code} ${req.text}')
```
Comment (dagnaldo) [inline core/views.py]: maximum 79 chars

File: core/views.py [hunk 45]
```diff
@@ -0,0 +1,262 @@
...
+
+            if req.json()['errors']:
+                insert_errors.extend([{'error': i['create']['error']['reason'], 'caused': i['create']
```
Comment (dagnaldo) [inline core/views.py]: maximum 79 chars

File: core/views.py [hunk 46]
```diff
@@ -0,0 +1,262 @@
...
+                                       ['error']['caused_by']} for i in req.json()['items'] if i['create']['status'] == 400])
+
+        print(f'[{datetime.now() - now}] Sent')
```
Comment (dagnaldo) [inline core/views.py]: check `now`

File: core/views.py [hunk 47]
```diff
@@ -0,0 +1,262 @@
...
+
+        return insert_errors
```
Comment (dagnaldo) [inline core/views.py]: change to `insertion_errors`?

File: docker-compose.yaml [hunk 48]
```diff
@@ -0,0 +1,69 @@
...
+      driver: none
+    environment:
+      - discovery.type=single-node
+      - "http.cors.allow-origin=*"
+      - "http.cors.enabled=true"
+      - "http.cors.allow-headers=X-Requested-With,X-Auth-Token,Content-Type,Con\
+        tent-Length,Authorization"
+      - "http.cors.allow-credentials=true"
+
+    network_mode: "host"
```
Comment (dagnaldo) [inline docker-compose.yaml]: `network_mode` is being used for?

File: docker-compose.yaml [hunk 49]
```diff
@@ -0,0 +1,69 @@
...
+      - .:/home/app/
+    environment:
+      DB_USER: defaultuser
+      DB_NAME: testdb
+      DB_PASSWORD: testpassword
+      DB_HOST: db
+      ALLOWED_HOSTS: '["*"]'
+    depends_on:
+      - es
+  # prod:
```
Comment (dagnaldo) [inline docker-compose.yaml]: remove, please

File: docker-compose.yaml [hunk 50]
```diff
@@ -0,0 +1,69 @@
...
+
+    network_mode: "host"
+    # ports:
+    #   - "9200:9200"
+    #   - "9300:9300"
+
+  dev:
+    build: .
+    command: >
+      bash -c " python manage.py makemigrations && python manage.py migrate
```
Comment (dagnaldo) [inline docker-compose.yaml]: Separate lines by &&

File: docker-compose.yaml [hunk 51]
```diff
@@ -0,0 +1,69 @@
...
+    volumes:
+      - .:/app/
+    network_mode: "host"
+    depends_on:
+      - es
+
+  test:
+    build: .
+    command: bash -c "export && python manage.py migrate && python manage.py test
+      core.tests.test_views.TestDetection"
```
Comment (dagnaldo) [inline docker-compose.yaml]: separate lines by &&

File: entrypoint.sh [hunk 52]
```diff
@@ -0,0 +1,18 @@
+#!/bin/bash
```
Comment (dagnaldo) [inline entrypoint.sh]: Unused file.

File: requirements.txt [hunk 53]
```diff
@@ -0,0 +1,9 @@
+django==3.1.7
+gunicorn==20.0.4
+djangorestframework==3.12.2
+djangorestframework-gis==0.17
+pandas==1.3.4
+pyyaml==5.2
+requests==2.19.1
+google-cloud-storage==1.36.2
+model-mommy~=1.5.1
```
Comment (dagnaldo) [inline requirements.txt]: model-mommy used only for tests, please create multiple tests files on `./requirements/`.

File: service_load_data_elasticsearch/settings.py [hunk 54]
```diff
@@ -0,0 +1,142 @@
...
+# Django Rest Framework
+# https://github.com/encode/django-rest-framework/
+
+REST_FRAMEWORK = {
+    'DEFAULT_RENDERER_CLASSES': [
+        'rest_framework.renderers.JSONRenderer',
+    ]
+}
+
+BULK_SIZE_REQUEST = os.getenv("BULK_SIZE_REQUEST")
```
Comment (dagnaldo) [inline service_load_data_elasticsearch/settings.py]: plase, provide a default data using `os.getenv("ENVIRONMENT", "DEFAULT_VALUE")`

**Review by dagnaldo — CHANGES_REQUESTED** [review]
Some changes requested is about docstrings and comments, please add `pylint` package on your vscode and try to save each file. It will automatically change your PEP8 errors or warnings and return a clean code to you.
Feel free to call me if u have any questions.

File: Dockerfile (hunk 1 above)
Comment (caiocacastro) [reply Dockerfile]: You mean every comment? Or just the entrypoint part?

File: core/tests/test_views.py (hunk 9 above)
Comment (caiocacastro) [reply core/tests/test_views.py]: It tests if a json file with a wrong formatting is sent and an error returns. I'm going to alter the test description.

File: service_load_data_elasticsearch/settings.py (hunk 54 above)
Comment (caiocacastro) [reply service_load_data_elasticsearch/settings.py]: Should I then send the default value to localhost url? Homolog version?

Comment (dagnaldo) [reply service_load_data_elasticsearch/settings.py]: Yes, please.

File: Dockerfile (hunk 1 above)
Comment (dagnaldo) [reply Dockerfile]: Every comment.

File: core/tests/test_views.py (hunk 9 above)
Comment (dagnaldo) [reply core/tests/test_views.py]: Yes, I know. But, I'm talking about the `response = {}`, is it used for something?

File: Dockerfile (hunk 1 above)
Comment (caiocacastro) [reply Dockerfile]: Done f9add46

File: core/models.py (hunk 3 above)
Comment (caiocacastro) [reply core/models.py]: Done f9add46

File: core/models.py (hunk 4 above)
Comment (caiocacastro) [reply core/models.py]: Done f9add46

File: core/serializers.py (hunk 5 above)
Comment (caiocacastro) [reply core/serializers.py]: Done f9add46

File: core/serializers.py (hunk 6 above)
Comment (caiocacastro) [reply core/serializers.py]: Done f9add46

File: core/tests/recipes.py (hunk 7 above)
Comment (caiocacastro) [reply core/tests/recipes.py]: Done f9add46

File: core/serializers.py (hunk 2 above)
Comment (caiocacastro) [reply core/serializers.py]: Done f9add46

File: core/tests/test_views.py (hunk 8 above)
Comment (caiocacastro) [reply core/tests/test_views.py]: Done f9add46

File: core/tests/test_views.py (hunk 9 above)
Comment (caiocacastro) [reply core/tests/test_views.py]: Done f9add46

File: core/tests/recipes.py (hunk 10 above)
Comment (caiocacastro) [reply core/tests/recipes.py]: Done f9add46

File: core/tests/test_views.py (hunk 11 above)
Comment (caiocacastro) [reply core/tests/test_views.py]: Done f9add46

File: core/tests/test_views.py (hunk 12 above)
Comment (caiocacastro) [reply core/tests/test_views.py]: Done f9add46

File: core/tests/test_views.py (hunk 13 above)
Comment (caiocacastro) [reply core/tests/test_views.py]: Done f9add46

File: core/tests/test_views.py (hunk 14 above)
Comment (caiocacastro) [reply core/tests/test_views.py]: Done f9add46

File: core/tests/test_views.py (hunk 15 above)
Comment (caiocacastro) [reply core/tests/test_views.py]: Done f9add46

File: core/urls.py (hunk 16 above)
Comment (caiocacastro) [reply core/urls.py]: Done f9add46

File: core/urls.py (hunk 17 above)
Comment (caiocacastro) [reply core/urls.py]: Done f9add46

File: core/views.py (hunk 19 above)
Comment (caiocacastro) [reply core/views.py]: Done f9add46

File: core/views.py (hunk 18 above)
Comment (caiocacastro) [reply core/views.py]: Done f9add46

File: core/views.py (hunk 20 above)
Comment (caiocacastro) [reply core/views.py]: Done f9add46

File: core/views.py (hunk 22 above)
Comment (caiocacastro) [reply core/views.py]: Done f9add46

File: core/views.py (hunk 21 above)
Comment (caiocacastro) [reply core/views.py]: Done f9add46

File: core/views.py (hunk 23 above)
Comment (caiocacastro) [reply core/views.py]: Done f9add46

File: core/views.py (hunk 24 above)
Comment (caiocacastro) [reply core/views.py]: Done f9add46

File: core/views.py (hunk 25 above)
Comment (caiocacastro) [reply core/views.py]: Done f9add46

File: core/views.py (hunk 26 above)
Comment (caiocacastro) [reply core/views.py]: Done f9add46

File: core/views.py (hunk 27 above)
Comment (caiocacastro) [reply core/views.py]: Done f9add46

File: core/views.py (hunk 28 above)
Comment (caiocacastro) [reply core/views.py]: Done f9add46

File: core/views.py (hunk 29 above)
Comment (caiocacastro) [reply core/views.py]: Done f9add46

File: core/views.py (hunk 30 above)
Comment (caiocacastro) [reply core/views.py]: Done f7376ec

File: core/views.py (hunk 32 above)
Comment (caiocacastro) [reply core/views.py]: Done f7376ec

File: core/views.py (hunk 34 above)
Comment (caiocacastro) [reply core/views.py]: Done f7376ec

File: core/views.py (hunk 33 above)
Comment (caiocacastro) [reply core/views.py]: Done f7376ec

File: core/views.py (hunk 31 above)
Comment (caiocacastro) [reply core/views.py]: Done f7376ec

File: core/views.py (hunk 35 above)
Comment (caiocacastro) [reply core/views.py]: Done f7376ec

File: core/views.py (hunk 36 above)
Comment (caiocacastro) [reply core/views.py]: Done f7376ec

File: core/views.py (hunk 37 above)
Comment (caiocacastro) [reply core/views.py]: Done f7376ec

File: core/views.py (hunk 39 above)
Comment (caiocacastro) [reply core/views.py]: Done f7376ec

File: core/views.py (hunk 38 above)
Comment (caiocacastro) [reply core/views.py]: Done f7376ec

File: core/views.py (hunk 40 above)
Comment (caiocacastro) [reply core/views.py]: Done f7376ec

File: core/views.py (hunk 41 above)
Comment (caiocacastro) [reply core/views.py]: Done f7376ec

File: core/views.py (hunk 42 above)
Comment (caiocacastro) [reply core/views.py]: Done f7376ec

File: core/views.py (hunk 43 above)
Comment (caiocacastro) [reply core/views.py]: Done f7376ec

File: core/views.py (hunk 45 above)
Comment (caiocacastro) [reply core/views.py]: Done f7376ec

File: core/views.py (hunk 44 above)
Comment (caiocacastro) [reply core/views.py]: Done f7376ec

File: core/views.py (hunk 46 above)
Comment (caiocacastro) [reply core/views.py]: Done f7376ec

File: core/views.py (hunk 47 above)
Comment (caiocacastro) [reply core/views.py]: Done f7376ec

File: docker-compose.yaml (hunk 48 above)
Comment (caiocacastro) [reply docker-compose.yaml]: Done f7376ec

File: docker-compose.yaml (hunk 49 above)
Comment (caiocacastro) [reply docker-compose.yaml]: Done f7376ec

File: docker-compose.yaml (hunk 50 above)
Comment (caiocacastro) [reply docker-compose.yaml]: Done f7376ec

File: docker-compose.yaml (hunk 51 above)
Comment (caiocacastro) [reply docker-compose.yaml]: Done f7376ec

File: entrypoint.sh (hunk 52 above)
Comment (caiocacastro) [reply entrypoint.sh]: Done f7376ec

File: requirements.txt (hunk 53 above)
Comment (caiocacastro) [reply requirements.txt]: Done f7376ec

File: service_load_data_elasticsearch/settings.py (hunk 54 above)
Comment (caiocacastro) [reply service_load_data_elasticsearch/settings.py]: Done f7376ec

File: core/models.py [hunk 55]
```diff
@@ -0,0 +1,95 @@
...
+            colon character
+        """
+        if field == 'dt_cadastro':
+            value = value.replace(tzinfo=None)
+        elif field == 'geometry':
+            value = value.wkt
+
+        return f'"{field}": "{value}"'
+
+    class Meta:
```
Comment (dagnaldo) [inline core/models.py]: Reordering META Class.

File: compose/docker/entrypoint.sh [hunk 56]
```diff
@@ -0,0 +1,17 @@
+set -e
```
Comment (dagnaldo) [inline compose/docker/entrypoint.sh]: unused file

File: core/models.py [hunk 57]
```diff
@@ -0,0 +1,95 @@
...
+        managed = False
+
+
+class BasicElasticStructure(models.Model):
```
Comment (dagnaldo) [inline core/models.py:83]: Missing `__str__` and `metaclass`

File: core/serializers.py [hunk 58]
```diff
@@ -0,0 +1,57 @@
...
+        Args:
+            feature (object): geojson feature uploaded for service
+
+        Returns:
+            object: attribute object to be serialized.
+        """
+        attr = {
+            '_id': feature['properties']['id'],
+            'geometry': geos.GEOSGeometry(json.dumps(feature["geometry"])).wkt,
+            'tb_ciclo_monitoramento_id': feature['properties']['tb_ciclo_monitoramento_id'],
```
Comment (dagnaldo) [inline core/serializers.py]: Max 79 characters

**Review by dagnaldo — CHANGES_REQUESTED** [review]
Some changes and a new PEP8 error format:
- Missing new line at the end of every python files.

File: core/tests/test_views.py (hunk 9 above)
Comment (dagnaldo) [reply core/tests/test_views.py]: Check it, please.

File: core/models.py (hunk 55 above)
Comment (caiocacastro) [reply core/models.py]: Reordered 8c11732

File: compose/docker/entrypoint.sh (hunk 56 above)
Comment (caiocacastro) [reply compose/docker/entrypoint.sh]: Deleted 8c11732

File: core/serializers.py (hunk 58 above)
Comment (caiocacastro) [reply core/serializers.py]: Broke line into two 8c11732

File: core/models.py (hunk 57 above)
Comment (caiocacastro) [reply core/models.py:83]: Done an __str__ and ordering by url and identifier 1dc995a

File: core/tests/test_views.py (hunk 9 above)
Comment (caiocacastro) [reply core/tests/test_views.py]: Sorry, fellow compatriot! I though you were asking about the whole test, not the specific response declaration. Fixed 9d0176b

**Review by dagnaldo — COMMENTED** [review]
I will run code and check if tests passed;

**Review by dagnaldo — COMMENTED** [review]
Missing some docstrings in pep8.

**Review by dagnaldo — APPROVED** [review]
Well done, a lot of changes, but, it's approved.

**Pull request merged**